### DISCUSSION TITLE
Integrating OPA with Zitadel

### DISCUSSION THREAD

**Discussion by cloudcompute** [zitadel/zitadel#7808] [Q&A]
Hi

Is it possible to integrate [`Open Policy Agent`](https://github.com/open-policy-agent/opa) for fine grained authorization / permissions with Zitadel or use it as a plugin?

By the way does Zitadel implement ABAC, ReBAC of its own?

Thank you.

**Comment by hifabienne** [comment]
I do not have experience with OPA, so I am not sure if thats possible.
@livio-a @muhlemmer you might know more than me?

ZITADEL does not have an implementation of ABAC, ReBAC.

Can you explain a little about your use case?

**Comment by cloudcompute** [comment]
[This link](https://www.permit.io/blog/relationship-based-access-control-rebac-with-open-policy-agent-opa) explains what I am looking for.

Thank you for the response.
